### DISCUSSION TITLE
Feature request/ideas

### DISCUSSION THREAD

**Discussion by R0CKH0PP3R** [SoongNoonien/mpdevil#117] [Ideas]
I'm loving plattenalbum for searching and playing my local library, so thank you so much for the seamless experience. :)

For me, that's not my main use-case for mpd though. I often stream Tidal from BubbleUPnP to upmpdcli & mpd. Unfortunately, Tidal do not embed cover-art, so there's noting to see in plattenalbum bar the default disc image. I know this probably isn't a very common use-case, but I assume the end result of missing cover art is a much wider experienced phenomenon. So my first feature request would be an option to download cover art when unavailable locally.  There are several free APIs out there for it these days - even from the major streamers themselves (I've personally used Deezer and found it quite painless).

My second feature request would be a start/close/minimise to tray option, where a tooltip displaying the album cover art, current track details & progress bar would be presented on hover (like Sonata, if you ever used that). 

Keep up the good work! :)

**Comment by SoongNoonien** [comment]
Hi!

> For me, that's not my main use-case for mpd though. I often stream Tidal from BubbleUPnP to upmpdcli & mpd. Unfortunately, Tidal do not embed cover-art, so there's noting to see in plattenalbum bar the default disc image. I know this probably isn't a very common use-case, but I assume the end result of missing cover art is a much wider experienced phenomenon. So my first feature request would be an option to download cover art when unavailable locally. There are several free APIs out there for it these days - even from the major streamers themselves (I've personally used Deezer and found it quite painless).

This could be done but it's currently not on my todo list. Dealing with automatic lyrics fetching is complicated and error prone enough. But for lyrics there is not alternative right now.

> My second feature request would be a start/close/minimise to tray option, where a tooltip displaying the album cover art, current track details & progress bar would be presented on hover (like Sonata, if you ever used that).

This has been discussed in #89 already.
